### PR TITLE
chore: Address typing issues in less problematic `meltano.cli` modules

### PR DESCRIPTION
**Pull request by edgarrmondragon**

### CODE REVIEW

File: src/meltano/cli/cli.py [hunk 1]
```diff
@@ -56,7 +57,7 @@ def main(self, *args, **kwargs) -> t.NoReturn:
     # NOTE: This CLI option normalization applies to all subcommands.
     context_settings={"token_normalize_func": lambda x: x.replace("_", "-")},
 )
-@click.option("--log-level", type=click.Choice(LEVELS.keys()))
+@click.option("--log-level", type=click.Choice(tuple(LEVELS)))
```
Comment (edgarrmondragon) [inline src/meltano/cli/cli.py:60]: https://github.com/pallets/click/pull/2736

File: src/meltano/cli/schedule.py [hunk 2]
```diff
@@ -193,11 +205,17 @@ def _format_elt_list_output(entry: Schedule, session: Session) -> dict:
...
-@click.option("--format", type=click.Choice(("json", "text")), default="text")
+@click.option(
+    "--format",
+    "list_format",
+    type=click.Choice(("json", "text")),
+    default="text",
+)
 @click.pass_context
-def list(ctx, format):  # noqa: WPS125
+def list_schedulers(ctx: click.Context, list_format: str) -> None:  # noqa: C901
```
Comment (edgarrmondragon) [inline src/meltano/cli/schedule.py:218]:
```suggestion
def list_schedules(ctx: click.Context, list_format: str) -> None:  # noqa: C901
```

File: src/meltano/cli/cli.py [hunk 3]
```diff
@@ -43,7 +43,8 @@ def main(self, *args, **kwargs) -> t.NoReturn:
             args: Positional arguments for the Click group.
             kwargs: Keyword arguments for the Click group.
         """
-        return super().main(*args, windows_expand_args=False, **kwargs)
+        # TODO: Fix "Return statement in function which does not return"?
+        return super().main(*args, windows_expand_args=False, **kwargs)  # type: ignore[misc]
```
Comment (WillDaSilva) [inline src/meltano/cli/cli.py:47]: Why are we returning anything from this function? If the return value is significant, why not change the type signature?

**Review by WillDaSilva — APPROVED** [review]

File: src/meltano/cli/cli.py (hunk 3 above)
Comment (edgarrmondragon) [reply src/meltano/cli/cli.py]: Yeah, I don't know why that didn't occur to me 😅

1a9fba1b7

**Pull request merged**